src/core/thread/mle.cpp

```diff
@@ -439,6 +440,9 @@ otError Mle::Restore(void)
     }
 #endif
 
+    // Successful restore indicates the device was attached to some Thread Network.
+    mIsFreshBoot = false;
```

Comment:
I'm not clear on the purpose of this flag. Which of the following cases is this flag intended to capture?

1. Device has been factory reset, but has not yet attached to any Thread network since factory reset.

1. Device has been rebooted, but has not yet attached to any Thread network since last reboot.

Reply:
2. the second scenario.

Reply:
If case 2, should we be setting `mIsFreshBoot` to `false` here? Shouldn't we wait until the role has changed to Child, Router, or Leader?

Reply:
Please refer to above first conversion for the background. Not sure but is the variable name or the comment bringing the confusion?

Reply:
I think it would help if there was more detail on what this flag is intended to represent and how it is used. This is the only place where I see `mIsFreshBoot` set to `false`. Are there other places where `mIsFreshBoot` should be set to `false`?

Reply:
Sorry for the confusion, to be accurate, the flag was designed to represent `Device joined one Thread Network, it had network info stored, and device is rebooted/restarted now, it restores the network info from flash and tries to reattach` and `leaving the role check to the caller who cares for fresh boot or not`.

For the DUA reregistration usage example - `The Device reregisters its DUA due to a Device reboot/restart in a random time delay chosen in the interval [0, Reregistration Delay]` , it should not re-register if it has not become attached and got Thread Network Data with Backbone Router Service.

I had considered to check whether it is `reboot` in `DuaManager::Restore()`, however there might be no DadCounter stored if  the DUA Interface Identifier is manually specified by user. And another consideration to put it here is that in practical scenarios, there might be some application level transactions which may cause burst of traffic in short time once multiple devices reboot together( and I am thinking whether or not providing one otAPI so that developer may decide to randomize the time in their application accordingly...)

Currently it is designed to reset this flag only accord to whether or not there is valid network info stored when bring up, there should be no other places to reset this value in my mind so far.

Thoughts?

Reply:
How about renaming the flag to `mIsRestoring`?

`Mle::Restore()` is also called as part of `otInstanceInit()` and that may be called for reasons not related to reboot.

Reply:
I also think `IsRestoring` or may be even `HasRestored` are name for this... this is to indicate that device restored network ifno (dataset, etc) from non-volatile settings after OT initialization (device boot).

Reply:
I would prefer `HasRestored` a bit as the suffix of `ing` looks like something ongoing.
Will update.